spec/specialist_publisher/discarding_draft_spec.rb

```diff
@@ -28,10 +28,20 @@ def when_i_discard_it
     expect_discarded_draft_alert(title)
   end
 
-  def then_i_get_a_404_on_draft_gov_uk
-    reload_url_until_status_code(@url, 404, keep_retrying_while: 200)
-
-    visit(@url)
-    expect(page).to have_text("Not found")
+  def then_i_get_a_410_on_draft_gov_uk
+    # Keep retrying while 200, as the page might have not been
+    # discarded yet. Keep retrying while 404, as while the content
+    # store deletes the routes before deleting the content, the
+    # reloading of the routes by the router is asynchronous, so there
+    # is the opportunity to get a 404 when the router still sends the
+    # request to Specialist Publisher, and the Content Store no longer
+    # has the content.
```

Comment:
Thinking we haven't used comments elsewhere + the commit message conveys the same info.

Reply:
Well, the information I was trying to convey which isn't in the commit is the reasoning behind the two status codes to keep retrying on. Although it might be hard to see that as the part of the code is now not that close to the relevant comment?